packages/shared/lib/services/activity/activity.service.ts

```diff
@@ -265,30 +263,35 @@ export async function getTopLevelLogByEnvironment(
  * each keyed by its associated log ID.
  */
 export async function getLogMessagesForLogs(logIds: number[], environment_id: number): Promise<ActivityLogMessagesGrouped> {
-    if (!logIds.length) {
-        return [];
-    }
-
     try {
+        const limit = 1000;
+
+        // Rank Partition will create one group per activity_log_id
+        // and allow us to ORDER with a GROUP by and add a "pseudo limit"
         const query = `
-            SELECT activity_log_id, array_agg(row_to_json(_nango_activity_log_messages.*)) as messages
-            FROM nango._nango_activity_log_messages
-            WHERE activity_log_id = ANY(?)
+        SELECT
+            *
+        FROM (
+            SELECT
+                *,
+                RANK() OVER (PARTITION BY activity_log_id ORDER BY created_at DESC) AS rank
+            FROM
+                _nango_activity_log_messages) AS partition
```

Comment:
no a comment but a preference. I found CTE https://www.postgresql.org/docs/current/queries-with.html style queries easier to read than subqueries. what do you think?
```
WITH ranked AS (
   SELECT
      *,
      RANK() OVER (PARTITION BY activity_log_id ORDER BY created_at DESC) AS rank
   FROM _nango_activity_log_messages) AS partition
   WHERE
      activity_log_id IN (${logIds.map(() => '?').join(',')})
      AND environment_id = ${environment_id}
)
SELECT
* 
FROM ranked
WHERE partition.rank <= ${limit}
```

Reply:
I will look 👌🏻